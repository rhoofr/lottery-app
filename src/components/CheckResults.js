import React, { useContext, useEffect } from 'react';
import Axios from 'axios';
import { withRouter } from 'react-router-dom';

import DispatchContext from '../shared/context/DispatchContext';
import LoadingSpinner from '../shared/components/uielements/LoadingSpinner';

const CheckResults = props => {
  const appDispatch = useContext(DispatchContext);

  useEffect(() => {
    const axiosRequest = Axios.CancelToken.source();

    async function fetchResults() {
      try {
        const response = await Axios.get(`/checkresults`, {
          cancelToken: axiosRequest.token
        });

        if (response.data) {
          appDispatch({ type: 'resultsLoaded', data: response.data.results });
        }
      } catch (error) {
        console.log('There was a problem or the request was cancelled.', error);
      }
      props.history.push(`/`);
    }
    fetchResults();
    // Cleanup
    return () => {
      axiosRequest.cancel();
    };
  }, [appDispatch, props.history]);

  return (
    <div className='center'>
      <LoadingSpinner asOverlay />
    </div>
  );
};

export default withRouter(CheckResults);
