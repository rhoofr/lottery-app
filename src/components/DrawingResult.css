.highlightWinner,
.highlightBallMatched,
.highlightNumbersMatched {
  font-weight: 700;
  color: #1fa326;
}

.highlightNotMatched {
  font-weight: 700;
  color: red;
}
