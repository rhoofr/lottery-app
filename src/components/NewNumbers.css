#edit__numbers {
  margin: 0 5rem;
}

#edit__card--title {
  margin-top: 0;
}

#form-delete {
  margin-top: 1rem;
}

#form-save {
  margin-top: 1rem;
}

@media screen and (max-width: 45em) {
  #post-first,
  #post-second,
  #post-third,
  #post-fourth,
  #post-fifth,
  #post-ball {
    height: calc(2.25rem + 2px);
    padding: 0.375rem 0.75rem;
    font-size: 1rem;
    border-radius: 0.25rem;
  }

  #edit__numbers {
    margin: 0 0.2rem;
  }

  #endDate {
    /* Full-screen display */
  }
}
