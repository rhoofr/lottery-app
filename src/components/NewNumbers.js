import React, { useEffect, useRef } from 'react';
import { withRouter } from 'react-router-dom';
import { useImmerReducer } from 'use-immer';
import Axios from 'axios';
import 'primeflex/primeflex.css';
import { RadioButton } from 'primereact/radiobutton';
import { Calendar } from 'primereact/calendar';
import { Message } from 'primereact/message';
import { Toast } from 'primereact/toast';

import Page from '../shared/containers/Page';
import LoadingSpinner from '../shared/components/uielements/LoadingSpinner';
import newNumbersReducer, { initialState } from '../shared/reducers/newNumbers';
import './NewNumbers.css';

export function NewNumbers(props) {
  const toast = useRef(null);

  const [state, dispatch] = useImmerReducer(newNumbersReducer, initialState);

  const submitHandler = e => {
    e.preventDefault();
    dispatch({ type: 'GAME_RULES', value: state.game.value });
    dispatch({
      type: 'FIRST_RULES',
      value: state.first.value,
      game: state.game.value
    });
    dispatch({
      type: 'SECOND_RULES',
      value: state.second.value,
      game: state.game.value
    });
    dispatch({
      type: 'THIRD_RULES',
      value: state.third.value,
      game: state.game.value
    });
    dispatch({
      type: 'FOURTH_RULES',
      value: state.fourth.value,
      game: state.game.value
    });
    dispatch({
      type: 'FIFTH_RULES',
      value: state.fifth.value,
      game: state.game.value
    });
    dispatch({
      type: 'BALL_RULES',
      value: state.ball.value,
      game: state.game.value
    });
    dispatch({ type: 'STARTDATE_RULES', value: state.startDate.value });
    dispatch({ type: 'ENDDATE_RULES', value: state.endDate.value });
    dispatch({ type: 'SUBMIT_REQUEST' });
  };

  useEffect(() => {
    if (state.sendCount) {
      dispatch({ type: 'CREATE_NUMBERS_STARTED' });

      const axiosRequest = Axios.CancelToken.source();

      async function updatePost() {
        try {
          const response = await Axios.post(
            `/numbersplayed`,
            {
              game: state.game.value,
              first: state.first.value,
              second: state.second.value,
              third: state.third.value,
              fourth: state.fourth.value,
              fifth: state.fifth.value,
              ball: state.ball.value,
              startDate: state.startDate.value,
              endDate: state.endDate.value
            },
            {
              cancelToken: axiosRequest.token
            }
          );
          dispatch({ type: 'CREATE_NUMBERS_COMPLETED' });
          if (response.data.success) {
            props.history.push(`/numbersplayed`);
          } else {
            return toast.current.show({
              severity: 'warn',
              summary: 'New Numbers Error',
              detail: `Error saving new numbers`,
              life: 3000
            });
          }
        } catch (error) {
          dispatch({ type: 'CREATE_NUMBERS_COMPLETED' });
          if (error.response) {
            console.log(
              'Error saving new numbers: ',
              error.response.data.message
            );
            return toast.current.show({
              severity: 'error',
              summary: 'New Numbers Error',
              detail: `Error saving new numbers: ${error.response.data.message}`,
              life: 3000
            });
          } else {
            console.log('Error saving new numbers', error);
            return toast.current.show({
              severity: 'error',
              summary: 'New Numbers Error',
              detail: `Error saving new numbers ${error}`,
              life: 3000
            });
          }
        }
      }
      updatePost();
      // Cleanup
      return () => {
        axiosRequest.cancel();
      };
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [state.sendCount]);

  return (
    <React.Fragment>
      {state.isSaving && (
        <div className='center'>
          <LoadingSpinner asOverlay />
        </div>
      )}
      {!state.isSaving && (
        <Page title='Create New Numbers'>
          <Toast ref={toast} />
          <div className='card'>
            <h3 className='card__title'>ADD NEW NUMBERS</h3>
            {state.hasNumbersError ? (
              <Message
                severity='error'
                text='Numbers must be greater than zero and from lowest to highest'
              ></Message>
            ) : null}
            {state.hasDateError ? (
              <Message
                severity='error'
                text='Date(s) are invalid for lottery type'
              ></Message>
            ) : null}
            <form className='mt-sm-2' onSubmit={submitHandler}>
              <div className='form-group'>
                <h5>Choose Game</h5>
                <div className='p-formgroup-inline'>
                  <div className='p-field-checkbox'>
                    <RadioButton
                      inputId='game1'
                      name='game'
                      value='M'
                      onChange={e =>
                        dispatch({ type: 'GAME_CHANGE', value: e.target.value })
                      }
                      checked={state.game.value === 'M'}
                    />
                    <label htmlFor='game1'>Mega Millions</label>
                  </div>
                  <div className='p-field-checkbox'>
                    <RadioButton
                      inputId='game2'
                      name='game'
                      value='P'
                      onChange={e =>
                        dispatch({ type: 'GAME_CHANGE', value: e.target.value })
                      }
                      checked={state.game.value === 'P'}
                    />
                    <label htmlFor='game2'>Powerball</label>
                  </div>
                  {state.game.hasErrors && (
                    <div className='alert alert-danger small liveValidateMessage'>
                      {state.game.message}
                    </div>
                  )}
                </div>
              </div>

              <div className='p-fluid p-formgrid p-grid'>
                <div className='p-field p-col'>
                  <label htmlFor='post-first' className='text-muted mb-1'>
                    First number
                  </label>
                  <input
                    onChange={e =>
                      dispatch({ type: 'FIRST_CHANGE', value: e.target.value })
                    }
                    onBlur={e =>
                      dispatch({
                        type: 'FIRST_RULES',
                        value: e.target.value,
                        game: state.game.value
                      })
                    }
                    name='first'
                    value={state.first.value}
                    id='post-first'
                    className='form-control form-control-lg form-control-first'
                    type='number'
                    placeholder=''
                    autoComplete='off'
                  />
                  {state.first.hasErrors && (
                    <div className='alert alert-danger small liveValidateMessage'>
                      {state.first.message}
                    </div>
                  )}
                </div>
                <div className='p-field p-col'>
                  <label htmlFor='post-second' className='text-muted mb-1'>
                    Second number
                  </label>
                  <input
                    onChange={e =>
                      dispatch({ type: 'SECOND_CHANGE', value: e.target.value })
                    }
                    onBlur={e =>
                      dispatch({
                        type: 'SECOND_RULES',
                        value: e.target.value,
                        game: state.game.value
                      })
                    }
                    name='second'
                    value={state.second.value}
                    id='post-second'
                    className='form-control form-control-lg form-control-second'
                    type='number'
                    placeholder=''
                    autoComplete='off'
                  />
                  {state.second.hasErrors && (
                    <div className='alert alert-danger small liveValidateMessage'>
                      {state.second.message}
                    </div>
                  )}
                </div>
              </div>

              <div className='p-fluid p-formgrid p-grid'>
                <div className='p-field p-col'>
                  <label htmlFor='post-third' className='text-muted mb-1'>
                    Third number
                  </label>
                  <input
                    onChange={e =>
                      dispatch({ type: 'THIRD_CHANGE', value: e.target.value })
                    }
                    onBlur={e =>
                      dispatch({
                        type: 'THIRD_RULES',
                        value: e.target.value,
                        game: state.game.value
                      })
                    }
                    name='third'
                    value={state.third.value}
                    id='post-third'
                    className='form-control form-control-lg form-control-third'
                    type='number'
                    placeholder=''
                    autoComplete='off'
                  />
                  {state.third.hasErrors && (
                    <div className='alert alert-danger small liveValidateMessage'>
                      {state.third.message}
                    </div>
                  )}
                </div>
                <div className='p-field p-col'>
                  <label htmlFor='post-fourth' className='text-muted mb-1'>
                    Fourth number
                  </label>
                  <input
                    onChange={e =>
                      dispatch({ type: 'FOURTH_CHANGE', value: e.target.value })
                    }
                    onBlur={e =>
                      dispatch({
                        type: 'FOURTH_RULES',
                        value: e.target.value,
                        game: state.game.value
                      })
                    }
                    name='fourth'
                    value={state.fourth.value}
                    id='post-fourth'
                    className='form-control form-control-lg form-control-fourth'
                    type='number'
                    placeholder=''
                    autoComplete='off'
                  />
                  {state.fourth.hasErrors && (
                    <div className='alert alert-danger small liveValidateMessage'>
                      {state.fourth.message}
                    </div>
                  )}
                </div>
              </div>

              <div className='p-fluid p-formgrid p-grid'>
                <div className='p-field p-col'>
                  <label htmlFor='post-fifth' className='text-muted mb-1'>
                    Fifth number
                  </label>
                  <input
                    onChange={e =>
                      dispatch({
                        type: 'FIFTH_CHANGE',
                        value: e.target.value
                      })
                    }
                    onBlur={e =>
                      dispatch({
                        type: 'FIFTH_RULES',
                        value: e.target.value,
                        game: state.game.value
                      })
                    }
                    name='fifth'
                    value={state.fifth.value}
                    id='post-fifth'
                    className='form-control form-control-lg form-control-fifth'
                    type='number'
                    placeholder=''
                    autoComplete='off'
                  />
                  {state.fifth.hasErrors && (
                    <div className='alert alert-danger small liveValidateMessage'>
                      {state.fifth.message}
                    </div>
                  )}
                </div>
                <div className='p-field p-col'>
                  <label htmlFor='post-ball' className='text-muted mb-1'>
                    Ball number
                  </label>
                  <input
                    onChange={e =>
                      dispatch({ type: 'BALL_CHANGE', value: e.target.value })
                    }
                    onBlur={e =>
                      dispatch({
                        type: 'BALL_RULES',
                        value: e.target.value,
                        game: state.game.value
                      })
                    }
                    name='ball'
                    value={state.ball.value}
                    id='post-ball'
                    className='form-control form-control-lg form-control-ball'
                    type='number'
                    placeholder=''
                    autoComplete='off'
                  />
                  {state.ball.hasErrors && (
                    <div className='alert alert-danger small liveValidateMessage'>
                      {state.ball.message}
                    </div>
                  )}
                </div>
              </div>

              <div className='p-fluid p-formgrid p-grid'>
                <div className='p-field p-col'>
                  <label htmlFor='startDate'>Start Date</label>
                  <Calendar
                    id='startDate'
                    value={state.startDate.value}
                    onChange={e =>
                      dispatch({
                        type: 'STARTDATE_CHANGE',
                        value: e.target.value
                      })
                    }
                    onBlur={e =>
                      dispatch({
                        type: 'STARTDATE_RULES',
                        value: e.target.value,
                        startDate: state.startDate.value
                      })
                    }
                    showIcon
                  />
                </div>
                {state.startDate.hasErrors && (
                  <div className='alert alert-danger small liveValidateMessage'>
                    {state.startDate.message}
                  </div>
                )}
                <div className='p-field p-col'>
                  <label htmlFor='endDate'>End Date</label>
                  <Calendar
                    id='endDate'
                    value={state.endDate.value}
                    onChange={e =>
                      dispatch({
                        type: 'ENDDATE_CHANGE',
                        value: e.target.value
                      })
                    }
                    onBlur={e =>
                      dispatch({
                        type: 'ENDDATE_RULES',
                        value: e.target.value,
                        endDate: state.endDate.value
                      })
                    }
                    showIcon
                  />
                </div>
                {state.endDate.hasErrors && (
                  <div className='alert alert-danger small liveValidateMessage'>
                    {state.endDate.message}
                  </div>
                )}
              </div>

              <button
                className='btn btn-lg btn-primary'
                disabled={state.isSaving}
              >
                {state.isSaving ? 'Saving...' : 'Save Numbers'}
              </button>
            </form>
          </div>
        </Page>
      )}
    </React.Fragment>
  );
}

export default withRouter(NewNumbers);
