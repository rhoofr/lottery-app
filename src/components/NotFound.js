import React from 'react';
import { Link } from 'react-router-dom';
import Page from '../shared/containers/Page';

function NotFound() {
  return (
    <Page title='Page not found'>
      <div className='text-center'>
        <h2>Page cannot be found.</h2>
        <p className='lead text-muted'>
          You can always visit the <Link to='/'>home page</Link> to get a fresh
          start.
        </p>
      </div>
    </Page>
  );
}

export default NotFound;
