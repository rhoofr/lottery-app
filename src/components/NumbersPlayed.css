.numbersPlayed {
  margin-top: 4rem;
}

.actions {
  text-align: center !important;
  margin: 0 0.2rem;
  width: 120px;
}

.p-button-rounded.action-button {
  height: 1.9rem !important;
  width: 1.9rem;
}

.p-button-rounded.actions {
  height: 2rem !important;
  width: 2rem;
}

.allResultsChecked {
  background-color: #aaf0d1 !important;
}

.numbersPlayed--title__highlight {
  color: #aaf0d1;
  font-weight: 700;
}

.dialog-delete__content {
  margin-bottom: 1rem;
}

@media screen and (max-width: 45em) {
  .actions {
    text-align: left !important;
  }

  .p-button-rounded.actions {
    height: 1.7rem !important;
    width: 1.7rem;
  }

  .edit-ticket {
    width: 100% !important;
  }
}
