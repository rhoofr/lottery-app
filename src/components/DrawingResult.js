import React from 'react';
import { v4 as uuidv4 } from 'uuid';
import classNames from 'classnames';

import { formatDate } from '../shared/utils/date';
import { formatCurrency } from '../shared/utils/formatting';

import './DrawingResult.css';

const ballMatchedBodyTemplate = drawResult => {
  const matchedClassName = classNames({
    highlightBallMatched: drawResult.ballMatched
  });
  return <td className={matchedClassName}>{drawResult.ball}</td>;
};

const highlightWinner = drawResult => {
  const matchedClassName = classNames({
    highlightWinner: drawResult.currentWinnings > 0
  });
  return (
    <td className={matchedClassName}>
      {formatCurrency(drawResult.currentWinnings)}
    </td>
  );
};

const highlightNumberMatched = (drawResult, nbr) => {
  const matchedClassName = classNames({
    highlightNumbersMatched: drawResult.matchedNums.includes(nbr)
  });
  return <td className={matchedClassName}>{nbr}</td>;
};

function DrawingResult({ drawResult }) {
  // console.log(props);
  return (
    <React.Fragment>
      <tr key={uuidv4()}>
        <td>{drawResult.game === 'P' ? 'Powerball' : 'Mega Millions'}</td>
        {highlightNumberMatched(drawResult, drawResult.winningNums[0])}
        {highlightNumberMatched(drawResult, drawResult.winningNums[1])}
        {highlightNumberMatched(drawResult, drawResult.winningNums[2])}
        {highlightNumberMatched(drawResult, drawResult.winningNums[3])}
        {highlightNumberMatched(drawResult, drawResult.winningNums[4])}
        {ballMatchedBodyTemplate(drawResult)}
        <td>{formatDate(drawResult.drawDate)}</td>
        {highlightWinner(drawResult)}
      </tr>
    </React.Fragment>
  );
}

export default DrawingResult;
