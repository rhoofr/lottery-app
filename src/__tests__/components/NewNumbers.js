import React from 'react';
import { shallow } from 'enzyme';
import { NewNumbers } from '../../components/NewNumbers';

test('should set first value on input change', () => {
  // const value = 'New description';
  const wrapper = shallow(<NewNumbers />);
  expect(wrapper).toMatchSnapshot();

  wrapper.find('input[id="post-first"]').simulate('change', {
    target: {
      value: '12'
    }
  });
  expect(wrapper.find('input[id="post-first"]').prop('value')).toBe('12');
});

test('should call onSubmit prop for form submission', () => {
  const onSubmitSpy = jest.fn();
  const wrapper = shallow(<NewNumbers onSubmit={onSubmitSpy} />);
  // console.log(wrapper.find('form'));
  wrapper.find('form').simulate('submit', {
    preventDefault: () => {}
  });
  // expect(wrapper.state('error')).toBe('');
  // expect(onSubmitSpy).toHaveBeenCalled();
});
