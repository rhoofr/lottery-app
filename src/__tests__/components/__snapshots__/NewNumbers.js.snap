// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should set first value on input change 1`] = `
<Fragment>
  <Page
    title="Create New Numbers"
  >
    <Toast
      baseZIndex={0}
      className={null}
      id={null}
      onClick={null}
      onRemove={null}
      position="top-right"
      style={null}
    />
    <div
      className="card"
    >
      <h3
        className="card__title"
      >
        ADD NEW NUMBERS
      </h3>
      <form
        className="mt-sm-2"
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <h5>
            Choose Game
          </h5>
          <div
            className="p-formgroup-inline"
          >
            <div
              className="p-field-checkbox"
            >
              <RadioButton
                ariaLabelledBy={null}
                checked={false}
                className={null}
                disabled={false}
                id={null}
                inputId="game1"
                name="game"
                onChange={[Function]}
                required={false}
                style={null}
                tabIndex={null}
                tooltip={null}
                tooltipOptions={null}
                value="M"
              />
              <label
                htmlFor="game1"
              >
                Mega Millions
              </label>
            </div>
            <div
              className="p-field-checkbox"
            >
              <RadioButton
                ariaLabelledBy={null}
                checked={false}
                className={null}
                disabled={false}
                id={null}
                inputId="game2"
                name="game"
                onChange={[Function]}
                required={false}
                style={null}
                tabIndex={null}
                tooltip={null}
                tooltipOptions={null}
                value="P"
              />
              <label
                htmlFor="game2"
              >
                Powerball
              </label>
            </div>
          </div>
        </div>
        <div
          className="p-fluid p-formgrid p-grid"
        >
          <div
            className="p-field p-col"
          >
            <label
              className="text-muted mb-1"
              htmlFor="post-first"
            >
              First number
            </label>
            <input
              autoComplete="off"
              className="form-control form-control-lg form-control-first"
              id="post-first"
              name="first"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="number"
              value=""
            />
          </div>
          <div
            className="p-field p-col"
          >
            <label
              className="text-muted mb-1"
              htmlFor="post-second"
            >
              Second number
            </label>
            <input
              autoComplete="off"
              className="form-control form-control-lg form-control-second"
              id="post-second"
              name="second"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="p-fluid p-formgrid p-grid"
        >
          <div
            className="p-field p-col"
          >
            <label
              className="text-muted mb-1"
              htmlFor="post-third"
            >
              Third number
            </label>
            <input
              autoComplete="off"
              className="form-control form-control-lg form-control-third"
              id="post-third"
              name="third"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="number"
              value=""
            />
          </div>
          <div
            className="p-field p-col"
          >
            <label
              className="text-muted mb-1"
              htmlFor="post-fourth"
            >
              Fourth number
            </label>
            <input
              autoComplete="off"
              className="form-control form-control-lg form-control-fourth"
              id="post-fourth"
              name="fourth"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="p-fluid p-formgrid p-grid"
        >
          <div
            className="p-field p-col"
          >
            <label
              className="text-muted mb-1"
              htmlFor="post-fifth"
            >
              Fifth number
            </label>
            <input
              autoComplete="off"
              className="form-control form-control-lg form-control-fifth"
              id="post-fifth"
              name="fifth"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="number"
              value=""
            />
          </div>
          <div
            className="p-field p-col"
          >
            <label
              className="text-muted mb-1"
              htmlFor="post-ball"
            >
              Ball number
            </label>
            <input
              autoComplete="off"
              className="form-control form-control-lg form-control-ball"
              id="post-ball"
              name="ball"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=""
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="p-fluid p-formgrid p-grid"
        >
          <div
            className="p-field p-col"
          >
            <label
              htmlFor="startDate"
            >
              Start Date
            </label>
            <Calendar
              appendTo={null}
              ariaLabelledBy={null}
              autoZIndex={true}
              baseZIndex={0}
              className={null}
              clearButtonClassName="p-button-secondary"
              dateFormat={null}
              dateTemplate={null}
              disabled={false}
              disabledDates={null}
              disabledDays={null}
              footerTemplate={null}
              headerTemplate={null}
              hideOnDateTimeSelect={false}
              hourFormat="24"
              icon="pi pi-calendar"
              id="startDate"
              inline={false}
              inputClassName={null}
              inputId={null}
              inputStyle={null}
              keepInvalid={false}
              locale={null}
              maxDate={null}
              maxDateCount={null}
              minDate={null}
              monthNavigator={false}
              name={null}
              numberOfMonths={1}
              onBlur={[Function]}
              onChange={[Function]}
              onClearButtonClick={null}
              onFocus={null}
              onInput={null}
              onSelect={null}
              onTodayButtonClick={null}
              onViewDateChange={null}
              panelClassName={null}
              panelStyle={null}
              placeholder={null}
              readOnlyInput={false}
              required={false}
              selectOtherMonths={false}
              selectionMode="single"
              shortYearCutoff="+10"
              showButtonBar={false}
              showIcon={true}
              showMillisec={false}
              showOnFocus={true}
              showOtherMonths={true}
              showSeconds={false}
              showTime={false}
              showWeek={false}
              stepHour={1}
              stepMillisec={1}
              stepMinute={1}
              stepSecond={1}
              style={null}
              tabIndex={null}
              timeOnly={false}
              todayButtonClassName="p-button-secondary"
              tooltip={null}
              tooltipOptions={null}
              touchUI={false}
              value=""
              view="date"
              viewDate={null}
              yearNavigator={false}
            />
          </div>
          <div
            className="p-field p-col"
          >
            <label
              htmlFor="endDate"
            >
              End Date
            </label>
            <Calendar
              appendTo={null}
              ariaLabelledBy={null}
              autoZIndex={true}
              baseZIndex={0}
              className={null}
              clearButtonClassName="p-button-secondary"
              dateFormat={null}
              dateTemplate={null}
              disabled={false}
              disabledDates={null}
              disabledDays={null}
              footerTemplate={null}
              headerTemplate={null}
              hideOnDateTimeSelect={false}
              hourFormat="24"
              icon="pi pi-calendar"
              id="endDate"
              inline={false}
              inputClassName={null}
              inputId={null}
              inputStyle={null}
              keepInvalid={false}
              locale={null}
              maxDate={null}
              maxDateCount={null}
              minDate={null}
              monthNavigator={false}
              name={null}
              numberOfMonths={1}
              onBlur={[Function]}
              onChange={[Function]}
              onClearButtonClick={null}
              onFocus={null}
              onInput={null}
              onSelect={null}
              onTodayButtonClick={null}
              onViewDateChange={null}
              panelClassName={null}
              panelStyle={null}
              placeholder={null}
              readOnlyInput={false}
              required={false}
              selectOtherMonths={false}
              selectionMode="single"
              shortYearCutoff="+10"
              showButtonBar={false}
              showIcon={true}
              showMillisec={false}
              showOnFocus={true}
              showOtherMonths={true}
              showSeconds={false}
              showTime={false}
              showWeek={false}
              stepHour={1}
              stepMillisec={1}
              stepMinute={1}
              stepSecond={1}
              style={null}
              tabIndex={null}
              timeOnly={false}
              todayButtonClassName="p-button-secondary"
              tooltip={null}
              tooltipOptions={null}
              touchUI={false}
              value=""
              view="date"
              viewDate={null}
              yearNavigator={false}
            />
          </div>
        </div>
        <button
          className="btn btn-lg btn-primary"
          disabled={false}
        >
          Save Numbers
        </button>
      </form>
    </div>
  </Page>
</Fragment>
`;
